src/Transport/AzureServiceBusTransport.cs

```diff
@@ -1,23 +1,227 @@
+            get => topicName;
+            set
+            {
+                Guard.AgainstNullAndEmpty(nameof(TopicName), value);
+                topicName = value;
+            }
+        }
+        string topicName = "bundle-1";
+
+        /// <summary>
+        /// The maximum size used when creating queues and topics in GB.
+        /// </summary>
+        public int EntityMaximumSize
+        {
+            get => entityMaximumSize;
+            set
+            {
+                Guard.AgainstNegativeAndZero(nameof(EntityMaximumSize), value);
+                entityMaximumSize = value;
+            }
+        }
+        int entityMaximumSize = 5;
+
+        /// <summary>
+        /// Enables entity partitioning when creating queues and topics.
+        /// </summary>
+        public bool EnablePartitioning { get; set; }
+
+        /// <summary>
+        /// Specifies the multiplier to apply to the maximum concurrency value to calculate the prefetch count.
+        /// </summary>
+        public int PrefetchMultiplier
+        {
+            get => prefetchMultiplier;
+            set
+            {
+                Guard.AgainstNegativeAndZero(nameof(PrefetchMultiplier), value);
+                prefetchMultiplier = value;
+            }
+        }
+        int prefetchMultiplier = 10;
+
+        /// <summary>
+        /// Overrides the default prefetch count calculation with the specified value. 
+        /// </summary>
+        public int? PrefetchCount
+        {
+            get => prefetchCount;
+            set
+            {
+                if (value.HasValue)
+                {
+                    Guard.AgainstNegative(nameof(PrefetchCount), value.Value);
+                }
+
+                prefetchCount = value;
+            }
+
+        }
+        int? prefetchCount;
+
+        /// <summary>
+        /// Overrides the default time to wait before triggering a circuit breaker that initiates the endpoint shutdown procedure when the message pump cannot successfully receive a message.
+        /// </summary>
+        public TimeSpan TimeToWaitBeforeTriggeringCircuitBreaker
+        {
+            get => timeToWaitBeforeTriggeringCircuitBreaker;
+            set
+            {
+                Guard.AgainstNegativeAndZero(nameof(TimeToWaitBeforeTriggeringCircuitBreaker), value);
+                timeToWaitBeforeTriggeringCircuitBreaker = value;
+            }
+        }
+        TimeSpan timeToWaitBeforeTriggeringCircuitBreaker = TimeSpan.FromMinutes(2);
+
+        /// <summary>
+        /// Specifies a callback to customize subscription names.
+        /// </summary>
+        public Func<string, string> SubscriptionNamingConvention
+        {
+            get => subscriptionNamingConvention;
+            set
+            {
+                Guard.AgainstNull(nameof(SubscriptionNamingConvention), value);
+
+                // wrap the custom convention:
+                subscriptionNamingConvention = subscriptionName =>
```

Comment:
that's a good point. In that case I'd rather vote to just remove the wrapper completely anyway though. Because once we call the convention, we don't really know anymore whether this is the default convention or a user-provided convention. So I'd say we leave it for now as is and if this is causing issues, we should look at just removing the wrapper.